src/Microsoft.Identity.Client/Platforms/iOS/iOSBroker.cs

```diff
@@ -0,0 +1,224 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using Microsoft.Identity.Client.Core;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Utils;
+using UIKit;
+using Foundation;
+using System;
+using CoreFoundation;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.Exceptions;
+
+namespace Microsoft.Identity.Client.Platforms.iOS
+{
+    /// <summary>
+    /// Handles requests which invoke the broker. This is only for mobile (iOS and Android) scenarios.
+    /// </summary>
+    internal class iOSBroker : IBroker
+    {
+        private static SemaphoreSlim brokerResponseReady = null;
+
+        private static NSUrl brokerResponse = null;
+        private readonly ApplicationConfiguration brokerRequest;
+        private Dictionary<string, string> _brokerPayload;
+
+        private readonly ICoreLogger _logger;
+        private IServiceBundle _serviceBundle;
+
+        public iOSBroker(ICoreLogger logger)
+        {
+            _logger = logger;
+        }
+
+        public bool CanInvokeBroker
+        {
+            get
+            {
+                CoreUIParent uiParent = new CoreUIParent();
+                if (!brokerRequest.IsBrokerEnabled)
+                {
+                    return false;
+                }
+
+                var result = false;
+
+                if (brokerRequest.IsBrokerEnabled)
+                {
+                    uiParent.CallerViewController.InvokeOnMainThread(() =>
+                    {
+                        result = UIApplication.SharedApplication.CanOpenUrl(new NSUrl("msauthv2://"));
+                    });
+                }
+
+                return result;
+            }
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenUsingBrokerAsync(Dictionary<string, string> brokerPayload, IServiceBundle serviceBundle)
```

Comment:
It would be better to pass the ServiceBundle through the ctor - as it does not change.